src/actions/multiCallHooks.js

```diff
@@ -29,95 +28,114 @@ export const useSubmitRegistration = (registrationDependencies) => {
         },
     }), [token]);
 
-    useEffect(()=>{
-        let aborted = false;
-
-        (async ()=> {
-            if(typeof registrationDependencies !== "undefined"){
-                const {tutoringRegistrations, classRegistrations, payment} = registrationDependencies;
+    useEffect(() => {
+        const aborted = false;
+        (async () => {
+            if (registrationDependencies) {
+                let {tutoringRegistrations, classRegistrations, payment} = registrationDependencies;
                 try {
-                    const TutoringCourses = await Promise.all(tutoringRegistrations.map( ({newTutoringCourse}) =>
-                        instance.request(
-                            {
-                                'url':courseEndpoint,
-                                ...requestSettings,
-                                'data': newTutoringCourse,
-                                'method':'post',
-                            }
-                        )
-                    ));
+                    const TutoringCourses = await Promise.all(
+                        tutoringRegistrations.map(({newTutoringCourse}) =>
+                            instance.post(courseEndpoint, newTutoringCourse, requestSettings))
+                    );
                     dispatch({
-                        type: types.POST_COURSE_SUCCESSFUL,
-                        payload: TutoringCourses,
+                        "payload": TutoringCourses,
+                        "type": types.POST_COURSE_SUCCESSFUL,
                     });
-                    const tutoringEnrollments = tutoringRegistrations.map((tutoringReg,i) =>
-                        ({student: tutoringReg.student, course: TutoringCourses[i].data.id}));
-                    // filter out classes were we're adding additional enrollments (there should be a valid enrollment id)
+                    const tutoringEnrollments = tutoringRegistrations.map((tutoringReg, i) =>
+                        ({
+                            "course": TutoringCourses[i].data.id,
+                            "student": tutoringReg.student,
+                        }));
+                    // get existing enrollments involving the given student-course pairs
+                    let currEnrollments = await Promise.all(
+                        classRegistrations.map(({student, course}) => instance.get(
+                            `/course/enrollment/?student=${student}&course_id=${course}`,
+                            requestSettings
+                        ))
+                    );
+
+                    // filter for the ones that found matches
+                    currEnrollments = currEnrollments
```

Comment:
Sort of debating whether or not we should validate this, but currently on the frontend it's not possible to pay for a registration cart if nothing's been selected or if there's no sessions for an enrollment. perhaps as a future enhancement we can validate if there are any currEnrollments before we perform the filtering